### PR TITLE
Fix vsync issue

### PR DESCRIPTION
**Pull request by xiaowei-guan**
After the engine is shutdown, OnVsync no longer be called.

### CODE REVIEW

**Comment by xiaowei-guan** [comment]
https://github.com/flutter-tizen/flutter-tizen/issues/232

**Review by swift-kim — COMMENTED** [review]
TV emulator is not currently available for me so I haven't tested the change but it generally looks good to me.

File: shell/platform/tizen/tizen_vsync_waiter.cc [hunk 1]
```diff
@@ -99,13 +107,19 @@ TdmClient::TdmClient(FlutterTizenEngine* engine) {
...
+  std::lock_guard<std::mutex> lock(engine_mutex_);
   engine_ = engine;
 }
 
 TdmClient::~TdmClient() {
   DestroyTdm();
 }
 
+void TdmClient::OnEngineStop() {
+  std::lock_guard<std::mutex> lock(engine_mutex_);
```
[the remark this answers is not in the document]
Comment (xiaowei-guan) [reply shell/platform/tizen/tizen_vsync_waiter.cc:118]: Yes, I have removed lock when create TdmClient, and add lock when call OnVsync

File: shell/platform/tizen/tizen_vsync_waiter.cc [hunk 2]
```diff
@@ -27,13 +27,20 @@ TizenVsyncWaiter::TizenVsyncWaiter(FlutterTizenEngine* engine)
...
+  }
   Send(kMessageQuit, 0);
   if (vblank_thread_) {
     ecore_thread_cancel(vblank_thread_);
     vblank_thread_ = nullptr;
   }
 }
 
+void TizenVsyncWaiter::SetTdmClient(TdmClient* tdmClient) {
+  tdmClient_ = tdmClient;
```
[the remark this answers is not in the document]
Comment (xiaowei-guan) [reply shell/platform/tizen/tizen_vsync_waiter.cc]: Ok

**Comment by xiaowei-guan** [comment]
> TV emulator is not currently available for me so I haven't tested the change but it generally looks good to me.

I have tested 20 times, no OnVsync error.

**Review by swift-kim — APPROVED** [review]

**Pull request merged**